Tools/wasm/emscripten/__main__.py

```diff
@@ -195,11 +195,10 @@ def configure_emscripten_python(context, working_dir):
     assert (
         len(lib_dirs) == 1
     ), f"Expected a single lib.* directory in {python_build_dir}"
-    lib_dir = os.fsdecode(lib_dirs[0])
-    pydebug = lib_dir.endswith("-pydebug")
-    python_version = lib_dir.removesuffix("-pydebug").rpartition("-")[-1]
+    _, python_version, abiflags = os.fsdecode(lib_dirs[0]).rsplit('-', maxsplit=2)
     sysconfig_data = (
-        f"{emscripten_build_dir}/build/lib.emscripten-wasm32-{python_version}"
+        f"{emscripten_build_dir}/build/"
+        f"lib.emscripten-wasm32-emscripten-{python_version}-{abiflags}"
```

Comment:
This won't *quite* work - `pydebug` is used in a couple of places (e.g., L203) that isn't covered by this change. I agree using `abiflags` is a better way to handle it - but we should probably use abiflags to set `pydebug`.

Reply:
Uh, I missed that line, I meant to do the exact same change I did to the WASI script. Since I changed the format of `pybuilddir.txt`, which is where `sysconfig_data` points to, we shouldn't be adding `-pydebug`. My change to the `sysconfig_data` variable already sets it to the correct value, I just forgot to remove the line below in the commit 😅.

I don't remember seeing `pydebug` being used elsewhere, but I will check when I am back to the computer.

That said, the script is broken as-is, meaning must not be covered by the CI. We should probably fix that.

Reply:
> That said, the script is broken as-is, meaning must not be covered by the CI. We should probably fix that.

Correct - is isn't, and yes, we should. 

The machine that will be used for the Emscripten buildbot has been commissioned. Right now, Emscripten hard-crashes on a couple of tests when `-uall` is enabled, but once those are resolved, we should be able to add the Emscripten buildbot to the pool.